shell/platform/windows/cursor_handler.cc

```diff
@@ -45,9 +70,180 @@ void CursorHandler::HandleMethodCall(
     const auto& kind = std::get<std::string>(kind_iter->second);
     delegate_->UpdateFlutterCursor(kind);
     result->Success();
+  } else if (method.compare(kCreateCustomCursorMethod) == 0) {
+    const auto& arguments = std::get<EncodableMap>(*method_call.arguments());
+    auto buffer_iter =
+        arguments.find(EncodableValue(std::string(kCustomCursorBufferKey)));
+    if (buffer_iter == arguments.end()) {
+      result->Error(
+          "Argument error",
+          "Missing argument buffer while trying to customize system cursor");
+      return;
+    }
+    auto buffer = std::get<std::vector<uint8_t>>(buffer_iter->second);
+    auto width_iter =
+        arguments.find(EncodableValue(std::string(kCustomCursorWidthKey)));
+    if (width_iter == arguments.end()) {
+      result->Error(
+          "Argument error",
+          "Missing argument width while trying to customize system cursor");
+      return;
+    }
+    auto width = std::get<int>(arguments.at(width_iter->second));
+    auto height_iter =
+        arguments.find(EncodableValue(std::string(kCustomCursorHeightKey)));
+    if (height_iter == arguments.end()) {
+      result->Error(
+          "Argument error",
+          "Missing argument height while trying to customize system cursor");
+      return;
+    }
+    auto height = std::get<int>(height_iter->second);
+    auto hot_x_iter =
+        arguments.find(EncodableValue(std::string(kCustomCursorHotXKey)));
+    if (hot_x_iter == arguments.end()) {
+      result->Error(
+          "Argument error",
+          "Missing argument hotX while trying to customize system cursor");
+      return;
+    }
+    auto hot_x = std::get<double>(hot_x_iter->second);
+    auto hot_y_iter =
+        arguments.find(EncodableValue(std::string(kCustomCursorHotYKey)));
+    if (hot_y_iter == arguments.end()) {
+      result->Error(
+          "Argument error",
+          "Missing argument hotY while trying to customize system cursor");
+      return;
+    }
+    auto hot_y = std::get<double>(hot_y_iter->second);
+    HCURSOR cursor = GetCursorFromBuffer(buffer, hot_x, hot_y, width, height);
+    if (cursor == nullptr) {
+      result->Error("Argument error",
+                    "Argument must contains a valid rawBGRA bitmap");
+      return;
+    }
+    // push the cursor into the cache vector of this handler.
+    custom_cursors_.emplace_back(std::move(cursor));
+    int64_t key = custom_cursors_.size() - 1;
```

Comment:
We should probably not derive `key` from the size of the vector `custom_cursors_`. If a key is ever removed, then all following keys are invalidated, but the user (framework code) does not know it.

Reply:
I've replaced `vector` with `unordered_map`, which accepts a `<std::string, HCURSOR>` pair.